velox/exec/tests/utils/PlanBuilder.cpp

```diff
@@ -315,6 +315,49 @@ std::unique_ptr<common::Filter> makeNotEqualFilter(
   }
 }
 
+std::unique_ptr<common::Filter> makeInFilter(const core::TypedExprPtr& expr) {
+  auto queryCtx = core::QueryCtx::createForTest();
+  auto vector = toConstant(expr, queryCtx);
+  VELOX_CHECK_EQ(vector->typeKind(), TypeKind::ARRAY);
+
+  auto arrayVector = vector->valueVector()->as<ArrayVector>();
+  auto index = vector->as<ConstantVector<ComplexType>>()->index();
+  auto offset = arrayVector->offsetAt(index);
+  auto size = arrayVector->sizeAt(index);
+
+  auto elementType = arrayVector->type()->asArray().elementType();
+  switch (elementType->kind()) {
+    case TypeKind::BIGINT: {
+      auto elements = arrayVector->elements()->as<SimpleVector<int64_t>>();
+      std::vector<int64_t> values;
+      for (auto i = 0; i < size; i++) {
+        values.push_back(elements->valueAt(offset + i));
+      }
+      return common::test::in(values);
+    }
+    case TypeKind::INTEGER: {
+      auto elements = arrayVector->elements()->as<SimpleVector<int32_t>>();
+      std::vector<int64_t> values;
```

Comment:
nit: `int32_t`
I feel templating the case block via `TypeTraits<Kind>::NativeType;` is cleaner.

Reply:
@majetideepak Deepak, I changed this to 

```
  switch (elementType->kind()) {
    case TypeKind::TINYINT:
      return common::test::in(toInt64List<int16_t>(elements, offset, size));
    case TypeKind::SMALLINT:
      return common::test::in(toInt64List<int16_t>(elements, offset, size));
    case TypeKind::INTEGER:
      return common::test::in(toInt64List<int32_t>(elements, offset, size));
    case TypeKind::BIGINT:
      return common::test::in(toInt64List<int64_t>(elements, offset, size));
    case TypeKind::VARCHAR: {
      auto stringElements = elements->as<SimpleVector<StringView>>();
      std::vector<std::string> values;
      for (auto i = 0; i < size; i++) {
        values.push_back(stringElements->valueAt(offset + i).str());
      }
      return common::test::in(values);
    }
    default:
      VELOX_NYI(
          "Unsupported value type for 'in' filter: {}",
          elementType->toString());
  }
```

Would you take another look?